bin/adhoc-scripts/createPileupObjects.py

```diff
@@ -0,0 +1,179 @@
+    RequestHandler = None
+try:
+    from WMCore.Services.ReqMgrAux.ReqMgrAux import ReqMgrAux
+except ImportError:
+    print("WMCore/ReqMgrAux environment not found. Faking it")
+    ReqMgrAux = None
+
+
+class OptionParser():
+    """Class to parse the command line arguments"""
+
+    def __init__(self):
+        "User based option parser"
+        self.parser = argparse.ArgumentParser(prog='PROG')
+        self.parser.add_argument("--fin", action="store",
+                                 dest="fin", default="", help="Input file")
+        self.parser.add_argument("--fout", action="store",
+                                 dest="fout", default="",
+                                 help="Output json file with all the pileup configurations created")
+        self.parser.add_argument("--verbose", action="store",
+                                 dest="verbose", default=0, help="verbosity level")
+        self.parser.add_argument("--url", action="store",
+                                 dest="url", default="https://cmsweb-testbed.cern.ch",
+                                 help="Server url to fetch and create pileup configurations")
+        self.parser.add_argument("--inject", action="store_true",
+                                 dest="inject", default=False,
+                                 help="Inject pileup documents for real")
+
+
+def getPUSchema(pileupName, pileupDocs, logger):
+    """
+    Get a minimalistic pileup data schema, or return an
+    existent pileup object.
+
+    :param pileupName: string with the pileup name
+    :param pileupDocs: list of pileup documents
+    :return: a dictionary with the pileup configuration
+    """
+    for doc in pileupDocs:
+        if pileupName == doc["pileupName"]:
+            logger.warning("Reusing pileup configuration for pileup: %s", pileupName)
+            return doc
+    pileupDoc = {
+        'pileupName': "",
+        'pileupType': "",
+        'expectedRSEs': [],
+        'campaigns': [],
+        'active': False}
+    return pileupDoc
+
+
+def parseCampaigns(campDocs, logger):
+    """
+    Parse campaigns from WMCore and convert them into
+    Pileup documents, where:
+     * campaigns: is a union of all campaigns using a given pileup.
+     * expectedRSEs: is a union of all RSEs across different campaigns.
+    """
+    pileupDocs = []
+    for camp in campDocs:
+        # for each secondary dataset, create one pileup document
+        for puName, puRSEs in camp.get("Secondaries", {}).items():
+            puDoc = getPUSchema(puName, pileupDocs, logger)
+            puType = "premix" if puName.split("/")[-1] == "PREMIX" else "classic"
+            puDoc["pileupName"] = puName
+            puDoc["pileupType"] = puType
+            puDoc["expectedRSEs"].extend(puRSEs)
+            puDoc["campaigns"].append(camp["CampaignName"])
+            # make some of these values unique
+            puDoc["expectedRSEs"] = list(set(puDoc["expectedRSEs"]))
```

Comment:
it is better to replace to something like this:
```
puDoc["expectedRSEs"] = list(set(puDoc.get("expectedRSEs", [)))
```

Reply:
`expectedRSEs` is a mandatory pileup attribute and I'd rather not have any default values for it.